js/lib/BWReader.js

```diff
@@ -0,0 +1,569 @@
+					var isLeaf = ba[nodeOffset];
+					//padding 8 byte
+					var children = sa[(nodeOffset/2)+1];
+					var offset = nodeOffset+4;
+
+					while(children>0){
+						children--;
+						var chrom = "";
+
+						for(var j=0;j<bbi.bpTree.keySize;j++){
+							var c = ba[offset+j];
+							if(c!=0) chrom += String.fromCharCode(c);
+						}
+
+						offset+=bbi.bpTree.keySize;
+						var idx = (ba[offset+3]<<24) | (ba[offset+2]<<16) | (ba[offset+1]<<8) | (ba[offset+0]);
+						var len = (ba[offset + 7]<<24) | (ba[offset+6]<<16) | (ba[offset+5]<<8) | (ba[offset+4]);
+						offset+=8;
+
+						bbi.chroms[idx] = chrom;
+						bbi.lengths[idx] = len;
+					}
+
+				}
+
+				error ? callback(null,error) : readChromTreeLeaf(32);
+				readRTreeIndex();
+			});
+		}
+
+		function readRTreeIndex(){
+			getData(bbi.unzoomedIndexOffset, 48, function(d){
+				var ba = new Uint8Array(d);
+				var sa = new Uint16Array(d);
+				var la = new Uint32Array(d);
+				var magic = la[0];
+
+				var readRTreeHeader = function(){
+					var Rheader = {};
+					Rheader.blockSize = la[1];
+					Rheader.nItems = read64Bit(ba,8);
+					Rheader.chrIdxStart = la[4];
+					Rheader.baseStart = la[5];
+					Rheader.chrIdxEnd = la[6];
+					Rheader.baseEnd = la[7];
+					Rheader.endFileOffset = read64Bit(ba,32);
+					Rheader.nItemsPerSlot = la[10];
+
+					//read32Bit(ba,44); // padding 4  byte
+					return Rheader;
+				}
+
+				if(magic==IDXTREE_MAGIC){
+					bbi.Rheader = readRTreeHeader();
+					bbi.rootOffset = bbi.unzoomedIndexOffset+48;
+					bbi.getValues = getValues;
+					callback(bbi);
+				}else{
+					callback(null,"R-tree not found!");
+				}
+
+			});
+		}
+
+		function getRTreeNode(treedata, offset){
+			var ba = new Uint8Array(treedata);
+			var sa = new Uint16Array(treedata);
+			var la = new Uint32Array(treedata);
+
+			var node = {};
+			node.isLeaf = ba[offset];
+			node.children = sa[offset/2 +1];
+			offset += 4;
+
+			node.chrIdxStart = new Array(node.children);
+			node.baseStart = new Array(node.children);
+			node.chrIdxEnd = new Array(node.children);
+			node.baseEnd = new Array(node.children);
+			node.dataOffset = new Array(node.children);
+			node.x = {};
+
+			if(node.isLeaf){
+				node.x.size = new Array(node.children);
+			} else {
+				node.x.child = new Array(node.children);
+				node.x.child.fill(-1);
+			}
+
+			for(var i = 0; i< node.children; i++){
+				var lo = offset / 4;
+				node.chrIdxStart[i] = la[lo];
+				node.baseStart[i] = la[lo + 1];
+				node.chrIdxEnd[i] = la[lo + 2];
+				node.baseEnd[i] = la[lo + 3];
+				node.dataOffset[i] = read64Bit(ba, offset+16);
+				offset += 24;
+
+				if(node.isLeaf){
+					node.x.size[i] = read64Bit(ba, offset);
+					offset += 8;
+				}
+			}
+
+			return node;
+		}
+
+		function getValues(chrom, start, end, cb){
+			var vals = [];
+			var chromid = bbi.chroms.indexOf(chrom);
+			if(chromid == -1) chromid = bbi.chroms.indexOf("chr" + chrom);
+			if(chromid == -1) cb(null, "chrom not found");
```

Comment:
This needs to be ```cb([], "chrom not found")``` - currently it causes an error to be thrown in parseData if you go to a chromosome not present in the file